### DISCUSSION TITLE
Multi user installation on offline environment

### DISCUSSION THREAD

**Discussion by dunatotatos** [nextflow-io/nextflow#4946] [Q&A]
Multi user installation of nextflow is easy (simply make sure the binary is in a standard location, like `/usr/local/bin`, or on your software stack).

Offline installation for a single user is well described in many places (copy the binary and `$HOME/.nextflow` to the offline system).

Multi user installation on an offline system seems to work with `-all` package, but this option is sub-optimal. It is not possible to install non-core plugins, and #4330 seems to indicate that `-all` might even disappear.

I've tried setting `NXF_HOME` to a shared folder, but as indicated in [this thread](https://groups.google.com/g/nextflow/c/yWnKVXAO8gU/m/nm1JKkrGAwAJ), it is a bad idea. I then looked for a way to share only relevant folders (mostly `capsule` and `framework`), but not `tmp`, and `cache`, but could not find how to.

Is there a recommended (or even just working) way to install Nextflow with support of non-core plugins for multiple users in an offline environment?

**Comment by mattdton** [comment]
Better support for this use case would be really useful. If you copy the `capsule` and `framework` directories into `${APP_PATH}`, setting

```
CAPSULE_CACHE_DIR=${APP_PATH}/capsule
NXF_DIST=${APP_PATH}/framework
```

seems to work but it would be good to know how fragile this is.

**Comment by bentsherman** [comment]
Since [24.07.0-edge](https://github.com/nextflow-io/nextflow/releases/tag/v24.07.0-edge), the "all" release was replaced with a "dist" release which supports the use of third-party plugins. Hope this helps
